### PR TITLE
move each dep into its own Makefile

### PR DESCRIPTION
**Pull request by vtjnash**
no code changes for now, just trying to getting the split out of the way quickly

inspired by https://github.com/JuliaLang/julia/pull/13812, this is an initial step towards re-developing our management of deps into a micro package manager, with better dependency tracking and the possibility of shared state

### CODE REVIEW

File: deps/unwind.mk [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+OSXUNWIND_OBJ_TARGET := $(build_shlibdir)/libosxunwind.$(SHLIB_EXT)
+OSXUNWIND_OBJ_SOURCE := $(BUILDDIR)/libosxunwind-$(OSXUNWIND_VER)/libosxunwind.$(SHLIB_EXT)
+
+$(SRCDIR)/srccache/libosxunwind-$(OSXUNWIND_VER).tar.gz: | $(SRCDIR)/srccache
+	$(JLDOWNLOAD) $@ https://github.com/JuliaLang/libosxunwind/archive/v$(OSXUNWIND_VER).tar.gz
+
+$(BUILDDIR)/libosxunwind-$(OSXUNWIND_VER)/Makefile: $(SRCDIR)/srccache/libosxunwind-$(OSXUNWIND_VER).tar.gz
+	$(JLCHECKSUM) $<
+	mkdir -p $(BUILDDIR)
+	cd $(BUILDDIR) && $(TAR) xfz $<
```
Comment (tkelman) [inline deps/unwind.mk]: no code changes, huh?

**Comment by tkelman** [comment]
easy to check locally via:
```
cat deps/tools/git-external.mk deps/llvm.mk deps/libuv.mk deps/pcre.mk \
  deps/openlibm.mk deps/openspecfun.mk deps/dsfmt.mk deps/rmath-julia.mk \
  deps/objconv.mk deps/blas.mk deps/arpack.mk deps/fftw.mk deps/utf8proc.mk \
  deps/suitesparse.mk deps/unwind.mk deps/gmp.mk deps/mpfr.mk deps/patchelf.mk \
  deps/git.mk deps/virtualenv.mk deps/libgit2.mk >> deps/Makefile
git diff HEAD~ deps/Makefile
```

**Comment by vtjnash** [comment]
I sense trust issues :). But that'll be useful to make sure this doesn't pick up any rebase conflicts.

**Comment by tkelman** [comment]
This particular change is fine and harmless, but regarding

> towards re-developing our management of deps into a micro package manager, with better dependency tracking and the possibility of shared state

please note that substantial refactoring of the build system is **not** on the release milestone and there are much more pressing things to work on right now.

**Comment by vtjnash** [comment]
> please note that substantial refactoring of the build system is not on the release milestone and there are much more pressing things to work on right now.

I realize that. The build system is usually just a side-project for me when I'm stalled on other work (like waiting for CI).

**Pull request merged**